deepspeed/moe/sharded_moe.py

```diff
@@ -519,6 +520,10 @@ def forward(self, *input: Tensor, **kwargs: Any) -> Tensor:
         if self.wall_clock_breakdown:
             self.timers('falltoall').start()
 
+        if groups._get_expert_model_parallel_world_size() == 1:
+            # no slicing for experts means we need to drop tokens
```

Comment:
Can we have a bit more detailed explanation here? To me "no expert slicing" has no direct relationship with "token dropping". And also good to explain how it's different from the drop_tokens in the gating functions in the same file.